CoreAlgorithmPlugins/test/unit/src/au/gov/asd/tac/constellation/plugins/algorithms/clustering/infomap/InfoMapPluginNGTest.java

```diff
@@ -192,42 +153,25 @@ private Config.ConnectionType connectionTypeHelper(final String connectionType)
         };
     }
 
-    private PluginParameters createParametersHelper(final String connectionType, final String dynamicType) {
-        final PluginParameters parameters = new PluginParameters();
+    private PluginParameters createParametersHelper(final InfoMapPlugin infoMapInstance, final String connectionType, final String dynamicType) {
+        final PluginParameters parameters = infoMapInstance.createParameters();
+        final PluginParameters updatedParameters = new PluginParameters();
 
         // Connection type
         final PluginParameter<SingleChoiceParameterType.SingleChoiceParameterValue> connectionParam = SingleChoiceParameterType.build(CONNECTION_TYPE_PARAMETER_ID);
         connectionParam.setName(CONNECTION_TYPE_PARAMETER_ID_NAME);
         SingleChoiceParameterType.setOptions(connectionParam, CONNECTION_TYPE_PARAM_VALUES);
         SingleChoiceParameterType.setChoice(connectionParam, connectionType);
-        parameters.addParameter(connectionParam);
+        updatedParameters.addParameter(connectionParam);
```

Comment:
is this different to the parameter for the original plugin? If not, the correct thing to do here would be to simply get the parameter and update the value rather than creating a new instance of it entirely. e.g.
```
connectionParam = parameters.getParameters().get(CONNECTION_TYPE_PARAMETER)
SingleChoiceParameterType.setChoice(connectionParam, connectionType)
```
If the value you're setting is the same as the default, you don't even need to change the value after calling instance.createParameters() (in which case you can probably remove the helper function here)

Reply:
Done. I originally thought it couldn't be done this way as getParameters returns an unmodifiable map, but I was mistaken about how that works. I've removed the helper function and streamlined that part of testing